### PR TITLE
Fix PE-D bugs

### LINKED ISSUE #229: [PE-D][Tester B] Add scheme gives wrong error message for very big scheme index number
Expected: The error message should not be shown when placing in an income that is very big integer
Actual: Error message shown
Steps to reproduce:
1. Input command: `addscheme 1 i/100000000000000000000000000000000000000000000000000000000000`
2. Error message shown `Index is not a non-zero unsigned integer.` although it is a non-zero unsigned integer.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: Virusrwj223/ped#13

### LINKED ISSUE #249: [PE-D][Tester C] Grammatical error for displaying eligible schemes format in the UG
At the 'Displaying eligible schemes: scheme' section, the format should be written as `scheme INDEX` rather than `scheme INDEXES` since only one user's index can be listed. The current `scheme INDEXES` format could potentially mislead users into thinking they can display the schemes which numerous people/indexes are eligible for (eg. `addscheme 1 3 2`), affecting the UG's clarity.

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: kikokidayo/ped#16

### PR DESCRIPTION
**Pull request by TerenceWongsExtravaganza**
This pull request does the following:

  1. Fix incorrect error message in addscheme command
  2. Fix grammatical error in User Guide relating to Scheme command
  3. Update Storage, Model and Better Model class diagram in Developer guide to reflect Scheme class
  4. Fix bug where duplicate index in deletescheme command causes display to show the same scheme getting deleted multiple times

Closes #229, #249

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -334,13 +334,17 @@ Format: `statistics`
 
 Displays the schemes that specified people from SocialBook are eligible for.
```
Comment (arjun2598) [inline docs/UserGuide.md]: Just some minor typos to point out: All the references to "people" should be changed to "person" and "indexes" to "index".

File: docs/UserGuide.md [hunk 2]
```diff
@@ -334,13 +334,17 @@ Format: `statistics`
...
 
-Format: `scheme INDEXES`
+Format: `scheme INDEX`
 
-* Displays the schemes that the people at the specified `INDEXES` are eligible for.
+* Displays the schemes that the people at the specified `INDEX` are eligible for.
 * The indexes refer to the index numbers shown in the displayed person list.
 * The indexes **must be a positive integer** 1, 2, 3, …​
```
Comment (arjun2598) [inline docs/UserGuide.md]: For this line, perhaps specifying that the index must not exceed the last displayed person's position would be better?

**Review by arjun2598 — COMMENTED** [review]
LGTM! Just some points to highlight

File: docs/UserGuide.md (hunk 2 above)
Comment (TerenceWongsExtravaganza) [reply docs/UserGuide.md]: thanks for pointing it out!

File: docs/UserGuide.md (hunk 1 above)
Comment (TerenceWongsExtravaganza) [reply docs/UserGuide.md]: aite will do

File: docs/UserGuide.md [hunk 3]
```diff
@@ -412,15 +412,19 @@ Format: `statistics`
 
 ### Displaying eligible schemes: `scheme`
 
-Displays the schemes that specified people from SocialBook are eligible for.
+Displays the schemes of the specified person in the address book is eligible for.
```
Comment (LeongWZ) [inline docs/UserGuide.md]: Not SocialBook instead of address book?

**Review by LeongWZ — APPROVED** [review]
LGTM! Highlighted an issue with the use of the term "address book" instead of "SocialBook" in the UG.

**Pull request merged**